apiserver/facades/client/client/status.go

```diff
@@ -644,6 +617,45 @@ func fetchRelations(ctx context.Context, relationService RelationService,
 	return out, outById, nil
 }
 
+func fetchOffers(ctx context.Context, service CrossModelRelationService) (map[string]offerStatus, error) {
+	modelOffers, err := service.GetOffers(ctx, []crossmodelrelation.OfferFilter{{}})
+	if err != nil {
+		return nil, err
+	}
+
+	return transform.SliceToMap(modelOffers, func(in *crossmodelrelation.OfferDetail) (string, offerStatus) {
+		charmURL, err := charms.CharmURLFromLocator(in.CharmLocator.Name, in.CharmLocator)
+
+		endpoints := transform.SliceToMap(in.Endpoints, func(in crossmodelrelation.OfferEndpoint) (string, charm.Relation) {
+			return in.Name, charm.Relation{
+				Name:      in.Name,
+				Role:      charm.RelationRole(in.Role),
+				Interface: in.Interface,
+				Limit:     in.Limit,
+			}
+		})
+		out := offerStatus{
+			ApplicationOffer: crossmodel.ApplicationOffer{
+				OfferUUID:              in.OfferUUID,
+				OfferName:              in.OfferName,
+				ApplicationName:        in.ApplicationName,
+				ApplicationDescription: in.ApplicationDescription,
+				Endpoints:              endpoints,
+			},
+			err:                  err,
+			charmURL:             charmURL,
+			activeConnectedCount: in.TotalActiveConnections,
+			totalConnectedCount:  in.TotalConnections,
+		}
+		return in.OfferName, out
+	}), nil
+}
+
+func fetchRemoteApplicationOfferers(ctx context.Context, service StatusService) (map[string]statusservice.RemoteApplicationOfferer, error) {
+	remoteAppOfferers, err := service.GetRemoteApplicationOffererStatuses(ctx)
+	return remoteAppOfferers, internalerrors.Capture(err)
+}
```

Comment:
Is there even a need for this method, seems pointless?

Reply:
Not particularly, just following the pattern